dotcom-rendering/src/components/ShareButton.importable.tsx

```diff
@@ -0,0 +1,229 @@
+`;
+
+const hoverStyles = css`
+	:hover {
+		color: ${themePalette('--share-button-liveblog-mobile')};
+		background-color: ${palette.neutral[100]};
+		svg {
+			fill: ${themePalette('--share-button-liveblog-mobile')};
+		}
+	}
+`;
+
+const liveBlogMobile = (isCopied: boolean) => css`
+	${until.desktop} {
+		color: ${palette.neutral[100]};
+		border-color: ${transparentColour(palette.neutral[100], 0.4)};
+		svg {
+			fill: ${isCopied ? palette.success[500] : palette.neutral[100]};
+		}
+		${!isCopied && hoverStyles}
+	}
+`;
+
+const getUrl = ({
+	pageId,
+	CMP,
+	blockId,
+}: {
+	pageId: string;
+	CMP?: string;
+	blockId?: string;
+}) => {
+	const searchParams = new URLSearchParams({});
+	if (CMP) searchParams.append('CMP', CMP); // Do we want to track this?
+	if (blockId) searchParams.append('page', `with:block-${blockId}`);
+
+	const blockHash = blockId ? `#block-${blockId}` : '';
+	const paramsString = searchParams.toString();
+	return new URL(
+		`${pageId}${paramsString ? '?' + paramsString : ''}${blockHash}`,
+		'https://www.theguardian.com/',
+	).href;
+};
+
+export const NativeShareButton = ({
+	onShare,
+	size,
+	isLiveBlogArticleMeta,
+}: {
+	onShare: () => void;
+	size?: Size;
+	isLiveBlogArticleMeta: boolean;
+}) => {
+	const sizeXSmall = size === 'xsmall';
+
+	return (
+		<Button
+			onClick={onShare}
+			size={size}
+			type="button"
+			priority="tertiary"
+			iconSide="left"
+			icon={<SvgShare />}
+			css={[
+				buttonStyles,
+				nativeShare(sizeXSmall),
+				isLiveBlogArticleMeta && liveBlogMobile(false),
+			]}
+		>
+			Share
+		</Button>
+	);
+};
+
+export const CopyLinkButton = ({
+	onShare,
+	size,
+	isLiveBlogArticleMeta,
+	isCopied,
+}: {
+	onShare: () => void;
+	size?: Size;
+	isLiveBlogArticleMeta: boolean;
+	isCopied: boolean;
+}) => {
+	const sizeXSmall = size === 'xsmall';
+	return (
+		<Button
+			onClick={onShare}
+			size={size}
+			type="button"
+			priority="tertiary"
+			iconSide="left"
+			icon={isCopied ? <SvgCheckmark /> : <LinkIcon />}
+			css={[
+				...(isCopied
+					? [copiedButtonStyles(sizeXSmall)]
+					: [buttonStyles]),
+				sharedButtonStyles(sizeXSmall, isCopied),
+				isLiveBlogArticleMeta && liveBlogMobile(isCopied),
+			]}
+		>
+			{isCopied ? 'Link copied' : 'Copy link'}
+		</Button>
+	);
+};
+
+export const ShareButton = ({
+	size = 'small',
+	pageId,
+	blockId,
+	webTitle,
+	format,
+	context,
+}: Props) => {
+	const [isCopied, setIsCopied] = useState(false);
+
+	const isLiveBlogArticleMeta =
+		format.design === ArticleDesign.LiveBlog && context === 'ArticleMeta';
+
+	const shareData = {
+		title: webTitle,
+		text: webTitle,
+		url: getUrl({
+			pageId,
+			blockId,
+		}),
+	};
+
+	useEffect(() => {
+		if (!isCopied) return;
+		const timer = setTimeout(() => {
+			setIsCopied(false);
+		}, 3000);
+		return () => clearTimeout(timer);
+	}, [isCopied]);
+
+	return typeof window === 'undefined' || typeof navigator === 'undefined' ? (
+		<></>
```

Comment:
I think it would be beneficial to think about what we want to render on the server and when a user does not have JavaScript.

Reply:
This is a good point, we may run into a CLS issue if we render nothing, although I'm not sure what the alternative might be.

Reply:
Reserve empty space?